### ISSUE TITLE
CPython difference for __contains__ method

### ISSUE THREAD

**Issue by stinos** [micropython/micropython#7884]
```
class TestContains:
  def __contains__(self, arg):
    return arg

print(0 in TestContains())
print(1 in TestContains())
print('' in TestContains())
print(0 not in TestContains())
print(1 not in TestContains())
print('' not in TestContains())
```

CPython:
```
False
True
False
True
False
True
```

because https://docs.python.org/3/reference/expressions.html#membership-test-operations

> For user-defined classes which define the __contains__() method, x in y returns True if y.__contains__(x) returns a true value, and False otherwise.

MicroPython:
```
0
1

True
False
True
```

Unexpected results so definitely worthy of a cpydiff addition, or should this be fixed?

**Comment by dlech** [comment]
Why does `print(2 in TestContains())` print `False` and an empty string? I would expect it to print `True` and `2`.

**Comment by stinos** [comment]
> Why does `print(2 in TestContains())` print `False` and an empty string? I would expect it to print `True` and `2`.

Sorry, because I pasted a previous version of the test code :) Fixed.

**Comment by dpgeorge** [comment]
I think this should be fixed, because it can lead to incorrect logic.

**Comment by dpgeorge** [comment]
Fixed by 1ded8a2977a84777f339ea7e7c788b11d75a6cce
